internal/jobobject/iocp.go

```diff
@@ -0,0 +1,106 @@
+package jobobject
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"unsafe"
+
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/queue"
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	ioInitOnce sync.Once
+	initIOErr  error
```

Comment:
Could there be a race between when initIOErr is set within the do once and when another thread comes in and attempts to check the initIOErr in Create?

Reply:
Yes we spoke on this, I'll add a mutex for this as Amit brought up a good point about how we're adding to the jobMap also in Create with no synchronization.

Reply:
I don't think there is a race on `initIOErr`. The `Once` ensures its call completes before any code after it is executed, so we should know `initIOErr` has been set before it is read.

Reply:
That's true, left as is

Reply:
Ah okay, golang spec clears that up, should be good. 

Spec for reference: https://golang.org/pkg/sync/#Once.Do